inject-dist/observers.js

```diff
@@ -21,14 +22,25 @@ const OBSERVERS = [
   'document.write',
   'queueMicrotask',
   'requestIdleCallback',
-  'navigator.scheduling.isInputPending',
-  'scheduler.postTask'
+  'scheduler.postTask',
+  'window.matchMedia',
+  'navigator.scheduling.isInputPending'
 ];
 
 const PROPERTIES_TO_TRACE = new Set([
   'navigator.userAgent'
 ]);
 
+// for each observer: custom function to determine which part of the argument should be captured
+const FUNCTION_CALL_ARGUMENTS_TO_CAPTURE = {
+  "window.matchMedia": function (arg) {
+    const m = arg.match(/\(([^:)]+)[:)]/);
+    if (m) {
+      return m[1];
+    }
+    return null;
+  }
```

Comment:
A few style/naming nits
```suggestion
  'matchMedia': function (mediaQueryString) {
    const match = mediaQueryString.match(/\(([^:)]+)[:)]/);
    if (match) {
      return match[1];
    }
    return null;
  }
```